test_all_notebooks.py

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python3
+
+'''Run all Jupyter Notebooks and check for errors.'''
+
+import os
+import subprocess
+import shutil
+import sys
+
+# TODO: fix the blacklisted notebooks
+BLACKLIST = [
+    'copernicus_dataspace.ipynb',
+    'data_usage.ipynb',
+    'layers.ipynb',
+    'permissions.ipynb',
+    'plots.ipynb',
+    'raster_dataset.ipynb',
+]
+
+
+def eprint(*args, **kwargs):
+    '''Print to stderr.'''
+    print(*args, file=sys.stderr, **kwargs)
+
+
+def run_test_notebook(notebook_path) -> bool:
+    '''Run test_notebook.py for the given notebook.'''
+
+    python_bin = shutil.which('python3')
+
+    if python_bin is None:
+        raise RuntimeError('Python 3 not found')
+
+    result = subprocess.run(
+        [python_bin, 'test_notebook.py', notebook_path],
+        capture_output=True,
+        text=True,
+        check=False,
+    )
+    if result.returncode == 0:
+        eprint(f"Notebook {notebook_path} ran successfully.")
+        return True
+    else:
+        eprint(f"Error running notebook {notebook_path}:")
+        eprint(result.stderr)
+        return False
+
+
+def main() -> int:
+    '''Run all Jupyter Notebooks and check for errors.'''
+
+    example_folder = 'examples'
+
+    if not os.path.isdir(example_folder):
+        eprint(f"The folder {example_folder} does not exist.")
+        return -1
+
+    for root, _dirs, files in os.walk(example_folder):
```

Comment:
run `n` in parallel?

Reply:
Ne, die Examples gehen alle auf :3030

Reply:
it would be possible to override this, if we adjust the examples to get the port from environment and only otherwise use 3030. We can postpone it but at some point I think running the examples sequentially will take too long.

Reply:
Das ginge, möchte ich aber gerade nicht machen, wenn es nicht notwendig ist.